server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1463,4 +2157,171 @@ protected ClusterBlockException checkBlock(UpdateIndexShardSnapshotStatusRequest
+
+        private final String repository;
+
+        FailPendingRepoTasksTask(String repository, Exception failure) {
+            this.repository = repository;
+            this.failure = failure;
+        }
+
+        @Override
+        public ClusterState execute(ClusterState currentState) {
+            final SnapshotDeletionsInProgress deletionsInProgress =
+                    currentState.custom(SnapshotDeletionsInProgress.TYPE, SnapshotDeletionsInProgress.EMPTY);
+            boolean changed = false;
+            final List<SnapshotDeletionsInProgress.Entry> remainingEntries = deletionsInProgress.getEntries();
+            List<SnapshotDeletionsInProgress.Entry> updatedEntries = new ArrayList<>(remainingEntries.size());
+            for (SnapshotDeletionsInProgress.Entry entry : remainingEntries) {
+                if (entry.repository().equals(repository)) {
+                    changed = true;
+                    deletionsToFail.add(entry.uuid());
+                } else {
+                    updatedEntries.add(entry);
+                }
+            }
+            final SnapshotDeletionsInProgress updatedDeletions = changed ? SnapshotDeletionsInProgress.of(updatedEntries) : null;
+            final SnapshotsInProgress snapshotsInProgress =
+                currentState.custom(SnapshotsInProgress.TYPE, SnapshotsInProgress.EMPTY);
+            final List<SnapshotsInProgress.Entry> snapshotEntries = new ArrayList<>();
+            boolean changedSnapshots = false;
+            for (SnapshotsInProgress.Entry entry : snapshotsInProgress.entries()) {
+                if (entry.repository().equals(repository)) {
+                    // We failed to read repository data for this delete, it is not the job of SnapshotsService to
+                    // retry these kinds of issues so we fail all the pending snapshots
+                    snapshotsToFail.add(entry.snapshot());
+                    changedSnapshots = true;
+                } else {
+                    // Entry is for another repository we just keep it as is
+                    snapshotEntries.add(entry);
+                }
+            }
+            final SnapshotsInProgress updatedSnapshotsInProgress = changedSnapshots ? SnapshotsInProgress.of(snapshotEntries) : null;
+            return updateWithSnapshots(currentState, updatedSnapshotsInProgress, updatedDeletions);
+        }
+
+        @Override
+        public void onFailure(String source, Exception e) {
+            logger.debug(
+                    () -> new ParameterizedMessage("Failed to remove all snapshot tasks for repo [{}] from cluster state", repository), e);
+            failAllListenersOnMasterFailOver(e);
+        }
+
+        @Override
+        public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
+            logger.trace("Removed all snapshot tasks for repository [{}] from cluster state, now failing listeners", repository);
```

Comment:
should we log this at info level? This is an important event?

Reply:
Yea let's even do `WARN` here. The only way to run into this is when failing to load `RepositoryData`. This should be almost impossible with retries and caching of `RepositoryData` and if it happens at any sort of frequency it should be very visible.